pwndbg/gdblib/disasm/arch.py

```diff
@@ -755,5 +756,50 @@ def _common_generic_register_destination(
 
             instruction.annotation = f"{left.str} => {self._telescope_format_list(telescope_addresses, TELESCOPE_DEPTH, emu)}"
 
+    def _common_cmp_annotator_builder(
+        self, flags_register_name: str, char_to_separate_operands: str = "-"
+    ) -> Callable[[PwndbgInstruction, Emulator], None]:
+        """
+        Many architectures implement near-identical `CMP`-like instructions.
+
+        It takes two values, either subtracts, adds, or does some bit operation
+        with them to set values in the flag register.
+
+        To reduce code duplication, subclasses can use this function to create an annotator for CMP-like instructions.
+        """
+        FLAG_REG_NAME_DISPLAY = flags_register_name.upper()
+
+        def handler(instruction: PwndbgInstruction, emu: Emulator):
```

Comment:
Note: this function is re-created each time the upper function is called. Maybe we should have it as some kind of `__privatebutnotrealy_func` :x.